src/Web/Handler.hs

```diff
@@ -528,11 +736,59 @@ postRootR = do
 getOverviewR :: TheoryIdx -> TheoryPath -> Handler Html
 getOverviewR idx path = withTheory idx ( \ti -> do
   renderF <- getUrlRender
+  lptxt <- getLemmaPlaintext idx path
+  --traceM lPlaintext
   defaultLayout $ do
-    overview <- liftIO $ overviewTpl renderF ti path
+    overview <- liftIO $ overviewTpl renderF ti path lptxt
     setTitle (toHtml $ "Theory: " ++ get thyName (tiTheory ti))
     overview )
 
+getTheoryVerifyR :: TheoryIdx -> TheoryPath -> Handler RepJson
+getTheoryVerifyR  idx (TheoryProof l path) = do
+    idx' <- editProof idx l
+    renderUrl <- getUrlRender
+    case idx' of
+        Right i -> return $ RepJson $ toContent $ object ["redirect" .= (renderUrl $ OverviewR i (TheoryProof l path))]
+        Left _  -> getTheoryPathMR idx TheoryHelp
+
+getTheoryVerifyR idx _ = do getTheoryPathMR idx TheoryHelp
+
+-- | Handles theory editing requests. 
+-- Theory delete doesn't require parsing the lemma, and is thus handled separatly
+postTheoryEditR :: TheoryIdx -> TheoryPath -> Handler Html
+postTheoryEditR idx (TheoryDelete l) = do
+    idx' <- deleteLemma idx l
+    case idx' of
+        Right i -> redirect (OverviewR i TheoryHelp)
+        Left  e -> do setMessage $ toHtml e
+                      redirect (OverviewR idx (TheoryDelete l))
+
+
+postTheoryEditR idx path = do
+    mLemmaText <- lookupPostParam "lemma-text" 
+    let newlptxt = T.unpack $ fromMaybe "" mLemmaText
+    maudeSig <- withTheory idx $ \ti -> return $ get sigpMaudeSig . toSignaturePure . get thySignature $ tiTheory ti
+    let parsed = parsePlainLemma maudeSig newlptxt
```

Comment:
I don't think you need this variable definition (only used once and rather short).